flytekit/core/type_engine.py

```diff
@@ -242,6 +245,13 @@ def from_binary_idl(self, binary_idl_object: Binary, expected_python_type: Type[
         else:
             raise TypeTransformerFailedError(f"Unsupported binary format `{binary_idl_object.tag}`")
 
+    def from_generic_idl(self, generic: Struct, expected_python_type: Type[T]) -> Optional[T]:
```

Comment:
YES, but I don't think `add a flag to the type` is a good choice now, since it might require IDL changes, and we should be super careful about IDL changes.